attributes.go

```diff
@@ -108,3 +110,36 @@ func statusCodeAttribute(protocol string, serverErr error) attribute.KeyValue {
 	}
 	return codeKey.String("success")
 }
+
+func headerAttributes(protocol, eventType string, metadata http.Header, allowedKeys []string) []attribute.KeyValue {
+	var attributes []attribute.KeyValue
+	for _, allowedKey := range allowedKeys {
+		if val, ok := metadata[allowedKey]; ok {
+			keyValue := attribute.String(
+				formatHeaderAttributes(protocol, eventType, formatHeaderAttributeKey(allowedKey)),
+				formatHeaderAttributeValue(val),
+			)
+			attributes = append(attributes, keyValue)
+		}
+	}
+	return attributes
+}
+
+func formatHeaderAttributes(protocol, eventType, key string) string {
```

Comment:
Implemented formatting suggested in [here](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/rpc.md#grpc-request-and-response-metadata)

Reply:
Can you put the link in a comment in the source code, so future readers know that this isn't arbitrary?

Reply:
I want to wait until [this pr](https://github.com/open-telemetry/opentelemetry-specification/pull/3116) is merged to link to `connect_rpc` conventions instead of grpc ones

Reply:
resolved in 39ab387349f28a2519527f3565e649a7e06e4380